### PR TITLE
Feature/bookmark icon

### PR DESCRIPTION
**Pull request by taranveersingh93**
Description
- Replace the button to save/remove recipes with a bookmark icon
- The bookmark icon is placed on the top left
Screenshot 2023-05-22 at 2 28 48 PM

- When clicked, the icon turns to orange, matching other colors on the display.
Screenshot 2023-05-22 at 2 28 36 PM

- Fixed the function that makes the column data for the grid so that the display is not disoriented for single/double columns.
Screenshot 2023-05-22 at 2 29 51 PM

- Added comments in html where data is rendered upon load.
- Still passing all tests

### CODE REVIEW

**Comment by taranveersingh93** [comment]
[[extension] Move the favorite recipe button](https://trello.com/c/ZyyFqZMs/41-extension-move-the-favorite-recipe-button)

**Comment by taranveersingh93** [comment]
[Fix the layout for when only 1-2 cards show up.](https://trello.com/c/GnLQMWap/35-fix-the-layout-for-when-only-1-2-cards-show-up)

**Comment by taranveersingh93** [comment]
[Add alt text](https://trello.com/c/1ajtTyrT/47-add-alt-text)

File: src/domUpdates.js [hunk 1]
```diff
@@ -33,7 +33,9 @@ const makeRecipeColumnData = (data) => {
   const leftColumn = mappedRecipe.filter(recipe => recipe.column === 1);
   const centreColumn = mappedRecipe.filter(recipe => recipe.column === 2);
   const rightColumn = mappedRecipe.filter(recipe => recipe.column === 0);
-  return [leftColumn, centreColumn, rightColumn];
+  const allColumns = [leftColumn, centreColumn, rightColumn];
+  const filteredColumns = allColumns.filter(column => column.length);
+  return filteredColumns;
```
Comment (mbenfowler) [inline src/domUpdates.js:38]: verry nice

File: src/images/appear-bookmark.png
Comment (mbenfowler) [inline src/images/appear-bookmark.png:1]: Are all of these images used?

File: src/scripts.js [hunk 2]
```diff
@@ -3,8 +3,11 @@
 
 import './styles.css'
 import { closeRecipe, showRecipe, switchView, searchForRecipes, updateUserRecipes, toggleTagData, renderActiveTag, displayTaggedRecipes, updateRecipesFromModal } from './domUpdates';
-import { calculateRecipeCost, getIngredientAmounts, getInstructions } from './recipes'; 
-import './images/antipasti.png';
+import { calculateRecipeCost, getIngredientAmounts, getInstructions } from './recipes';
+import './images/appear-bookmark-icon.png'
+import './images/hover-bookmark-icon.png'
+import './images/select-bookmark-icon.png'
+import './images/antipasti.png'
```
Comment (mbenfowler) [inline src/scripts.js]: do these need to be imported here? Actually it looks like none of these images need to be imported here anymore?

**Review by mbenfowler — CHANGES_REQUESTED** [review]

**Review by lauraguerra1 — APPROVED** [review]

File: src/scripts.js (hunk 2 above)
Comment (taranveersingh93) [reply src/scripts.js]: It was asked for in the original document. I would not want to change that due to the approaching deadline. Can remove the unused pictures.

**Review by mbenfowler — APPROVED** [review]

**Pull request merged**